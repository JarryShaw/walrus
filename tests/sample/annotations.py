def foo1(x=(h1 := 1)):
    pass


print(h1)


def foo2(x: int = (h2 := 2)):
    pass


print(h2)


def foo3(x: (h3 := int)):
    pass


print(h3)


def foo4(x: (h4 := float) = 1.1): pass


print(h4)


def foo5(x=1) -> (h5 := complex):
    return x * 1j


print(h5)
