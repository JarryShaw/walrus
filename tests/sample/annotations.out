1
2
<class 'int'>
<class 'float'>
<class 'complex'>
